### ISSUE TITLE
too many arguments in call to files.NewSerialFile

### ISSUE THREAD

**Issue by NatoBoram** [ipfs/ipfs-cluster#618]
#### Pre-check

* [x] This is not a IPFS Cluster website content issue ([file those here](https://github.com/ipfs/ipfs-cluster-website/issues))
* [x] I read the [troubleshooting section](https://cluster.ipfs.io/documentation/troubleshooting/) of the website and it did not help
* [x] I searched for similar issues in the repo without luck
* [x] All my peers are running the same cluster version
* [x] All my peers are configured using the same cluster `secret`

#### Basic information

* [x] Version information (mark as appropiate):
  * [x] Master
  * [ ] Release candidate for next version
  * [ ] Latest stable version
  * [ ] An older version I should not be using
* [x] Type (mark as appropiate):
  * [x] Bug
  * [ ] Feature request
  * [ ] Enhancement
* [x] Operating system (mark as appropiate):
  * [x] Linux
  * [ ] macOS
  * [ ] Windows
  * [ ] Other: `which?`
* [x] Installation method (mark as appropiate):
  * [ ] Binaries from dist.ipfs.io
  * [x] Built from sources
  * [ ] Docker
  * [ ] Snap
  * [ ] Other: `which?`

#### Description

I was trying to read a bit in the source code but I found quite a few errors, and I wondered if it's normal. Here's just one of them.

On line [/api/rest/client/methods.go#L376](https://github.com/ipfs/ipfs-cluster/blob/master/api/rest/client/methods.go#L376), there's : 

```go
	return files.NewSerialFile(path.Base(fpath), fpath, params.Hidden, stat)
```

However, I've got the following error : 

```
../../api/rest/client/methods.go:376:28: too many arguments in call to files.NewSerialFile
        have (string, string, bool, os.FileInfo)
        want (string, bool, os.FileInfo)
```

Indeed, in the definition of `files.NewSerialFile`, which is [here](https://github.com/ipfs/go-ipfs-files/blob/master/serialfile.go#L35), the signature is different : 

```go
func NewSerialFile(path string, hidden bool, stat os.FileInfo) (Node, error) {
```

Is this something to be concerned about?

**Comment by hsanjuan** [comment]
I don't think you rewrote dependencies before building: https://cluster.ipfs.io/documentation/download/#installing-from-source
